### PR TITLE
EBNT-260 HTTP API reference for projects and tasks

### PR DESCRIPTION
**Pull request by i-a-andrianov**
The first MR in HTTP API reference series

### CODE REVIEW

File: docs/http-api/projects.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Get project
+-----------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  GET /projects/1
```
Comment (mike0sv) [inline docs/http-api/projects.rst]: specify that 1 is `id`

File: docs/http-api/projects.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  DELETE /projects/1?cascade=1
+
+* `cascade`: (optional, default - `0`) delete cascadely (with referenced tasks, models, etc)
+
+Response
+^^^^^^^^^^^^^^
+
+* `204`: OK
+* `400`: if `cascade` is not `1` and given project has tasks in it
+* `404`: if given project doesn't exist
```
Comment (mike0sv) [inline docs/http-api/projects.rst]: what about update?

File: docs/http-api/tasks.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Get list of tasks
+-----------------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  GET /tasks?project_id=1
```
Comment (mike0sv) [inline docs/http-api/tasks.rst]: I thought that we will have something like `/project/<id>/tasks`
like described [here](https://www.moesif.com/blog/technical/api-design/REST-API-Design-Best-Practices-for-Sub-and-Nested-Resources/)
if you have a link (or your own argument) that proves me wrong, i am all ears

Comment (i-a-andrianov) [reply docs/http-api/tasks.rst]: I like this thing. In fact earlier version of this document (made in January) was based on this idea. Unfortunately this had several downsides in our case:
1. URLs are too long: `/projects/.../tasks/.../models/.../images`;
1. Unclear what to do with instances which "reference" both images and environments;
1. More suited for work with natural ids (in our case names: `/projects/<name>/tasks/<name>`, etc) and not synthetic ones: to retrieve a model with some id we do not need ids of project and task it belongs to but in this case we should either pass them anyway or have too much URL variations.

File: docs/http-api/projects.rst (hunk 2 above)
Comment (i-a-andrianov) [reply docs/http-api/projects.rst]: This is really a good question. `Project` has `id`, `name`, `author` and `creation_date`. `Task` has the same attributes and `project_id`. I believe the only thing which could be safely updated here is `name`. But do we actually need it? I don't know.

File: docs/http-api/projects.rst (hunk 1 above)
Comment (i-a-andrianov) [reply docs/http-api/projects.rst]: fixed

File: docs/http-api/projects.rst (hunk 2 above)
Comment (mike0sv) [reply docs/http-api/projects.rst]: Potentially there will be more fields. For example, description?
In task it is almost certain - when we add datasets and metrics.
Also, if you look at it from admin UI perspective - yes, we may need a way to change all the fields (for example, move tasks between projects, changing projects author)

Comment (i-a-andrianov) [reply docs/http-api/projects.rst]: fixed

**Pull request merged**